sys/log/full/src/log_fcb_slot1.c

```diff
@@ -49,30 +49,45 @@ static bool g_slot1_locked;
         })
 
 static int
-log_fcb_slot1_append(struct log *log, void *buf, int len)
+log_fcb_slot1_append_body(struct log *log, struct log_entry_hdr *hdr,
+                          const void *body, int body_len)
 {
     return LOG_FCB_SLOT1_CALL(log, log_append, buf, len);
 }
 
 static int
-log_fcb_slot1_append_body(struct log *log, const struct log_entry_hdr *hdr,
-                          const void *body, int body_len)
+log_fcb_slot1_append(struct log *log, void *buf, int len)
 {
-    return LOG_FCB_SLOT1_CALL(log, log_append_body, hdr, body, body_len);
+    return log_fcb_slot1_append_body(log, (uint8_t *)buf + LOG_BASE_ENTRY_HDR_SIZE,
+                                     len - LOG_BASE_ENTRY_HDR_SIZE);
 }
 
 static int
-log_fcb_slot1_append_mbuf(struct log *log, const struct os_mbuf *om)
+log_fcb_slot1_append_mbuf_body(struct log *log,
+                               struct log_entry_hdr *hdr,
+                               struct os_mbuf *om)
 {
-    return LOG_FCB_SLOT1_CALL(log, log_append_mbuf, om);
+    return LOG_FCB_SLOT1_CALL(log, log_append_mbuf_body, hdr, om);
 }
 
 static int
-log_fcb_slot1_append_mbuf_body(struct log *log,
-                               const struct log_entry_hdr *hdr,
-                               const struct os_mbuf *om)
+log_fcb_slot1_append_mbuf(struct log *log, struct os_mbuf *om)
 {
-    return LOG_FCB_SLOT1_CALL(log, log_append_mbuf_body, hdr, om);
+    struct os_mbuf m;
+    uint16_t mlen;
+
+    mlen = os_mbuf_len(om);
+    if (mlen < LOG_BASE_ENTRY_HDR_SIZE) {
+        return SYS_ENOMEM;
+    }
+
+    memcpy(&m, om, sizeof m);
+
+    os_mbuf_pullup(om, LOG_BASE_ENTRY_HDR_SIZE);
+
+    m.om_data = m.om_data + log_hdr_len((struct log_entry_hdr *)om->om_data);
+
+    return log_fcb_slot1_append_mbuf_body(log, (struct log_entry_hdr *)om->om_data, &m);
```

Comment:
As with `log_cbmem_append_mbuf()` and `log_fcb_append_mbuf()`:
* Should use `OS_MBUF_PKTLEN()`
* Some comments would help a lot!

Reply:
Not applicable here.